### DISCUSSION TITLE
Default data not showing in UI

### DISCUSSION THREAD

**Discussion by shyam0118** [filamentphp/filament#9383] [General]
I have sent an email and  a page will open when I click on the button present inside the email and I want to show the default data and I can modify the data and I can save the data but the default data is not showing in the UI and the data is present in the mount function. Please help me to fix this...Thank you  and here is the code `class AppraisalResponseRequest extends EditRecord
{

    protected static string $resource = AppraisalResponseResource::class;

    protected static string $layout = 'components.request.layouts.app';
    public $emp;
    public $review_period_start_date;
    public $review_period_end_date;
    public $type;
    public $appraisal_session_id;

    protected function getRedirectUrl(): string
    {
        return url(route('open-appraisal.submitted'));
    }

    public function getTitle(): string
    {
        return strval(__('open-appraisal::open-appraisal.section.responses.request', ["employee_code" => $this->record->employee->employee_code]));
    }

    protected function authorizeAccess(): void
    {
    }

    public function mount(int | string $record = null): void
    {
        try {
            $data = JWT::decode(app('request')->input('token'), ['header' => true]);
        } catch (\Throwable $th) {
            abort(403);
        }
        $appraisal_response = AppraisalResponse::where([['id', $data['response_id']], ['status', AppraisalResponseStatus::DRAFT]])->first();
        // dd($appraisal_response);
        if (!$appraisal_response) {
            abort(404);
        }
        parent::mount($appraisal_response->id);
    }

    protected function getHeaderActions(): array
    {
        return [
            //
        ];
    }

    protected function afterSave(): void
    {
        $this->record->status = AppraisalResponseStatus::SUBMITTED();
        $this->record->save();
    }

    public function form(Form $form): Form
    {
        $form = static::$resource::form($form, true);
        return $form;
    }
}`   all data present inside $appraisal_response  and in the UI only the default value of review_period_start_date, review_period_end_date and type showing and the reset data not showing

**Comment by leandrocfe** [comment]
If you want to fill the form you can use `$this->form->fill(['field' => 'xx'])`

https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component/#initializing-the-form-with-data

Or using hooks($this->data['field'] = 'xx')

https://filamentphp.com/docs/3.x/panels/resources/creating-records/#lifecycle-hooks

**Reply by shyam0118** [reply]
Thanks for you response, And I tried this but when I try to set the default data using fill() it is not working and it is showing this error "Unable to find component: [app.filament.resources.appraisal-response-resource.pages.appraisal-response-request] and I can't configure out where the problem is occurring. Please help me to solve this issue. Thank you.

**Reply by leandrocfe** [reply]
Is it a custom page?

**Reply by shyam0118** [reply]
yes it is a custom page
